### ISSUE TITLE
Add assembly name to standard naming for annotation

### ISSUE THREAD

**Issue by petrelharp** [popsim-consortium/stdpopsim#1252]
We have the assembly name at the end of every genetic map; this should probably be the case for annotations also, so it's obvious if they don't match.

**Comment by nspope** [comment]
@andrewkern suggests adding a property to the class that carries the annotation name

**Comment by silastittes** [comment]
Happy to work on this. Which class specifically should carry the annotation name?

**Comment by petrelharp** [comment]
I would think [`Annotation`](https://popsim-consortium.github.io/stdpopsim-docs/latest/api.html#stdpopsim.Annotation); are there any other candidates, @silastittes?

**Comment by silastittes** [comment]
`Annotation` has an `id` field already, and the full file name is in the url field. What info would go in the annotation name?

**Comment by nspope** [comment]
That's my bad -- "annotation name" ==> "assembly name"

**Comment by silastittes** [comment]
Should an assembly name be added to genetic maps too then?

**Comment by petrelharp** [comment]
I think figuring that out is part of this issue.

**Comment by silastittes** [comment]
It does seem like it would be nice to be able to do things like 
```
assert genetic_map.assembly == annotation.assembly
```

**Comment by petrelharp** [comment]
Ping @silastittes - might you be able to figure this out (in the next week)?

**Comment by silastittes** [comment]
Yeah happy to work on this.

**Comment by nspope** [comment]
This should hopefully fix #719

**Comment by petrelharp** [comment]
hey @silastittes - any progress on this?

**Comment by silastittes** [comment]
No! Sorry, other things including edits on the manuscript took priority. After reviewing conversation, I'm a bit confused. Is this about adding new attributes to classes or just having conventions for the file names? Both?

**Comment by silastittes** [comment]
Ok, re-re-reading ... We don't currently have a genome version attribute in Annotation, or GeneticMap, but we do in [Genome](https://github.com/popsim-consortium/stdpopsim/blob/d07f5e59cae2b02a2628111776f0797bd180cfb0/stdpopsim/genomes.py#L44). So a potential solution could be to just carry over the same formatting to the two other classes to facilitate checks that they match, and enforce naming conventions that include the genome assembly in map and annotation files. Does that cover the concerns?

**Comment by petrelharp** [comment]
Ah, okay - first off, many apologies; my issue was totally misguided. Here's what's going on, I think (please check?):

I was talking about the *files*, but neglected to say so. For these:

- Uploaded annotation files contain a version number (`_vX.tar.gz`) to disambiguate them (#1553)
- Uploaded genetic map files do not; however, their filenames contain the name of the assembly
- So, it's a reasonable question: do genetic map files need a version number in their name? Do annotation files need an assembly name in their name? I think the answer is "no", because both serve to disambiguate them. And IIRC we decided on "version number" rather than "assembly" for some reason.

Now, you reasonably interpreted my issue as asking something like "shouldn't we be able to know what assembly a given annotation is associated with". (and the answer to that is "yes"!) Currently we record that information in the `description`:
![Screenshot from 2025-01-09 15-29-46](https://github.com/user-attachments/assets/2ab47c33-c1bb-4b74-92af-cc687ebcca6a)
![Screenshot from 2025-01-09 15-29-33](https://github.com/user-attachments/assets/a2ed5286-4344-4390-8cab-7196d563794b)
... but it would be reasonable for it to be an attribute, to ensure this?  And, that's what your PR is doing (#1650). This is in fact necessary, since we haven't enforced the `description` thing:
![Screenshot from 2025-01-09 15-32-59](https://github.com/user-attachments/assets/663db6af-26e2-45ad-ae98-33be14c6907e)


Okay! I'm on board. Do you agree about the lack of an issue around filenames?
